content/master/concepts/compositions.md

```diff
@@ -29,385 +65,210 @@ Crossplane has four core components that users commonly mix up:
+
+This example uses
+[Function Patch and Transform]({{<ref "../guides/function-patch-and-transform">}}).
+Function Patch and Transform implements Crossplane resource
+templates.  
+The input kind is {{<hover label="single" line="13">}}Resources{{</hover>}}, 
+and it accepts  {{<hover label="single" line="14">}}resources{{</hover>}} as input.
 
-```yaml {label="controller1",copy-lines="none"}
+```yaml {label="single",copy-lines="none"}
 apiVersion: apiextensions.crossplane.io/v1
 kind: Composition
+# Removed for Brevity
 spec:
-  resources:
-    - base:
-        apiVersion: ec2.aws.upbound.io/v1beta1
-        kind: VPC
-        name: my-vpc
-        spec:
-          forProvider:
-          # Removed for brevity
-    - base:
-        apiVersion: ec2.aws.upbound.io/v1beta1
-        kind: InternetGateway
-        name: my-gateway
-        spec:
-          forProvider:
-            vpcIdSelector:
-              matchControllerRef: true
-# Removed for brevity
+  # Removed for Brevity
+  mode: Pipeline
+  pipeline:
+  - step: patch-and-transform
+    functionRef:
+      name: function-patch-and-transform
+    input:
+      apiVersion: pt.fn.crossplane.io/v1beta1
+      kind: Resources
+      resources:
+      - name: storage-bucket
+        base:
+          apiVersion: s3.aws.upbound.io/v1beta1
+          kind: Bucket
+          spec:
+            forProvider:
+              region: "us-east-2"
 ```
 
+### Use a pipeline of functions in a composition
 
-Resources can match both labels and a controller reference to match a specific
-resource in the larger composite resource. 
-
-For example, this Composition creates two 
-{{<hover label="controller2" line="17">}}VPC{{</hover>}}
-resources, but the 
-{{<hover label="controller2" line="27">}}InternetGateway{{</hover>}} 
-must match only one. 
+Crossplane can ask more than one Function what to do when a composite resource
+changes. When a Composition has a pipeline of two or more steps, Crossplane
+calls them all. It calls them in the order they appear in the pipeline.
 
-Applying a {{<hover label="controller2" line="21">}}label{{</hover>}} to the 
-second {{<hover label="controller2" line="17">}}VPC{{</hover>}} allows the 
-{{<hover label="controller2" line="27">}}InternetGateway{{</hover>}} to match
-the label
-{{<hover label="controller2" line="34">}}type: internet{{</hover>}} and only
-match objects in the same composite resource with 
-{{<hover label="controller2" line="32">}}matchControllerRef{{</hover>}}.
+Crossplane passes each Function in the pipeline the result of the previous
+Function. This enables powerful combinations of Functions. In this example,
+Crossplane calls {{<hover label="double" line="10">}}function-cue{{</hover>}} to
+create an S3 bucket. Crossplane then passes the bucket to 
+{{<hover label="double" line="23">}}function-auto-ready{{</hover>}}, which marks the
+composite resource as ready when the bucket becomes ready.
```

Comment:
nit: function-auto-ready could be confusing for someone learning Crossplane / compositions, e.g. why do I need to mark it to ready again with a separate function, if it is already ready :) It would demonstrate the idea better if the second function adds a common label/annotation etc.

Reply:
This content is pre-existing so I'm going to leave it as is.